catroid/src/main/java/org/catrobat/catroid/devices/arduino/ArduinoImpl.java

```diff
@@ -187,6 +187,10 @@ private void tryInitialize() throws IOException, SerialException {
 			sendFirmataMessage(new SetPinModeMessage(pin, SetPinModeMessage.PIN_MODE.PWM.getMode()));
 		}
 		reportSensorData(true);
+
+		// get status of digital ports
+		sendFirmataMessage(new ReportDigitalPortMessage(0, true));
+		sendFirmataMessage(new ReportDigitalPortMessage(1, true));
```

Comment:
Why do you send this firmata messages here?
Before reading a pin, the same firmata message is send anyway. (see [here](https://github.com/Catrobat/Catroid/blob/509c87e5041a5d173e439183c01b109b5e824726/catroid/src/main/java/org/catrobat/catroid/devices/arduino/ArduinoImpl.java#L262))